File: annealers/Object.py
class Object:
    def __init__(self, total_cost, 
                 cost_con_value, 
                 budget_constraint_val, 
                 truck_num_constraint_val, 
                 excavator_values, truck_values, 
                 half_used_values, 
                 total_revenue_val, 
                 obj_val, 
                 produce_cost, 
                 oil_consume_cost, 
                 maintenance_cost, 
                 precurement_cost, 
                 excavator_produce_dict): 
        self.total_cost = total_cost
        self.cost_con_value = cost_con_value
        self.budget_constraint_val = budget_constraint_val
        self.truck_num_constraint_val = truck_num_constraint_val
        self.excavator_values = excavator_values
        self.truck_values = truck_values
        self.half_used_values = half_used_values
        self.total_revenue_val = total_revenue_val
        self.obj_val = obj_val
        self.produce_cost = produce_cost
        self.oil_consume_cost = oil_consume_cost
        self.maintenance_cost = maintenance_cost
        self.precurement_cost = precurement_cost
        self.excavator_produce_dict = excavator_produce_dict
            

File: crates/data_dealer/src/object.rs
use std::{collections::BTreeMap, fs};

use serde::{Serialize, Deserialize};

#[derive(Debug,Serialize, Deserialize, Default, Clone)]
pub struct Object {
    pub total_cost: f32,
    pub cost_con_value: f32,
    pub truck_num_constraint_val : BTreeMap<String, f32>,
    pub excavator_values: BTreeMap<String, f32>,
    pub truck_values: BTreeMap<String, f32>,
    pub half_used_values: BTreeMap<String, f32>,
    pub total_revenue_val: f32,
    pub obj_val: f32,
    pub produce_cost: f32,
    pub oil_consume_cost: f32,
    pub maintenance_cost: f32,
    pub precurement_cost: f32,
    pub excavator_produce_dict: BTreeMap<String, f32>
}

impl Object {
    pub fn from_file(file_path: &str) -> Self{
        let str = fs::read_to_string(file_path).expect("Couldn't find or load that file.");
        let object : Self = serde_json::from_str(&str).unwrap();
        object
    }

    pub fn to_readable_string(&self) -> String {
        let mut string = String::new();
        string.push_str(&format!("total_cost is {}\n", self.total_cost));
        string.push_str(&format!("excavator_values is {:?}\n", self.excavator_values));
        string.push_str(&format!("truck_value is {:?}\n", self.truck_values));
        string.push_str(&format!("half_used_values is {:?}\n", self.half_used_values));
        string.push_str(&format!("total_revenue_val is {}\n", self.total_revenue_val));
        string
    }
}
